### PR TITLE
Add a fake remote settings endpoint to nimbus-cli server

### PR DESCRIPTION
**Pull request by jhugman**
Fixes [EXP-3718](https://mozilla-hub.atlassian.net/browse/EXP-3718).

### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

File: components/support/nimbus-cli/src/output/server.rs [hunk 1]
```diff
@@ -305,14 +348,67 @@ mod tests {
...
+
+        let s = get(port, "/buckets/BUCKET/collections/COLLECTION/records").await?;
+        assert_eq!(s, serde_json::to_string(&value)?);
+
+        let _ = tx.send(());
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_getting_null_values_from_fake_remote_settings() -> Result<()> {
```
Comment (eliserichards) [inline components/support/nimbus-cli/src/output/server.rs:391]: Do we need a test for the `(StatusCode::NOT_MODIFIED, Json(Value::Null))` case too?

**Review by jeddai — COMMENTED** [review]
So I ran into issues testing this — I spun up the server, ran enroll with `--pbcopy`, then ran `open` with `--pbpaste` and it _says_ it pushed it to the server but when I hit the collection endpoint I get the 304 for not modified

**Comment by jhugman** [comment]
> So I ran into issues testing this — I spun up the server, ran enroll with `--pbcopy`, then ran `open` with `--pbpaste` and it _says_ it pushed it to the server but when I hit the collection endpoint I get the 304 for not modified

Hmm. This may be a UX issue.

`--pbcopy` isn't that useful: it _just_ copies the URL into the desktop clipboard.
`--pbpaste` _just_ pushes the URL (and now the experiments) to the server.

`--pbcopy` and `--pbpaste` and conceptually linked, but there not paired. I would love the hear ideas of what else we could call this.

So, your `enroll --pbcopy` — copies a URL to the clipboard.
`open --pbpaste` sends a different URL to the server, but open has no experiments; so the `index.html` has a URL, but no experiments. The output of collection endpoint has nothing from the last command (i.e. the `open`), so returns a 304.

**Comment by jeddai** [comment]
Oh what should I be doing then instead of `open`?

**Comment by jhugman** [comment]
> Oh what should I be doing then instead of `open`?

```sh

# Start the server, may be better done in another terminal window.
nimbus-cli start-server &

# Any command that sends experiments to the device can use `--pbpaste` to send them via the server.
nimbus-cli --app firefox_ios --channel release enroll <EXPERIMENT> --branch <BRANCH> --pbpaste
```

This PR just makes the experiments available via the Remote Settings endpoint.

**Review by jeddai — APPROVED** [review]
That did it, thanks @jhugman!
